src/docfx/lib/markdown/ResolveXref.cs

```diff
@@ -15,31 +16,49 @@ internal static class ResolveXref
         {
             return builder.Use(document =>
-            {
-                document.Replace(node =>
-                {
-                    if (node is XrefInline xref)
-                    {
-                        var xrefSpec = resolveXref(xref.Href);
-                        if (xrefSpec is null)
-                        {
-                            var raw = xref.GetAttributes().Properties.First(p => p.Key == "data-raw-source").Value;
-                            var error = raw.StartsWith("@")
-                                ? Errors.AtUidNotFound((Document)InclusionContext.File, xref.Href, raw)
-                                : Errors.UidNotFound((Document)InclusionContext.File, xref.Href, raw);
+             {
+                 document.Replace(node =>
+                 {
+                     if (node is XrefInline xref)
+                     {
+                         var (uid, query, _) = HrefUtility.SplitHref(xref.Href);
+                         var xrefSpec = resolveXref(uid);
+                         if (xrefSpec is null)
+                         {
+                             var raw = xref.GetAttributes().Properties.First(p => p.Key == "data-raw-source").Value;
+                             var error = raw.StartsWith("@")
+                                 ? Errors.AtUidNotFound((Document)InclusionContext.File, xref.Href, raw)
+                                 : Errors.UidNotFound((Document)InclusionContext.File, xref.Href, raw);
 
-                            Markup.Result.Errors.Add(error);
-                            return new LiteralInline(raw);
-                        }
+                             Markup.Result.Errors.Add(error);
+                             return new LiteralInline(raw);
+                         }
 
-                        // TODO: Support advanced cross reference
-                        // e.g.: <a href="xref:System.String?displayProperty=fullName"/>
-                        var content = new LiteralInline(string.IsNullOrEmpty(xrefSpec.GetName()) ? xrefSpec.Uid : xrefSpec.GetName());
-                        return new LinkInline(xrefSpec.Href, null).AppendChild(content);
-                    }
-                    return node;
-                });
-            });
+                         string display;
+                         var content = string.IsNullOrEmpty(xrefSpec.GetName()) ? xrefSpec.Uid : xrefSpec.GetName();
+                         if (!string.IsNullOrEmpty(query))
+                         {
+                             var queries = HttpUtility.ParseQueryString(query.Substring(1));
+                             var displayProperty = queries["displayProperty"];
+                             if (!string.IsNullOrEmpty(displayProperty))
+                             {
+                                 display = xrefSpec.GetXrefPropertyValue(displayProperty);
```

Comment:
display could be null if the property does not exists, suggest to use the following fallback:

```csharp

propertyName = query?.displayName ?? "name"
display = xrefspec.ExtensionData[propertyName] ?? uid

```